src/util.ts

```diff
@@ -42,6 +40,34 @@ export function getUserAgent(): string {
   return ((appIdentifier.length > 0) ? `${appIdentifier} ` : '') + baseUserAgent;
 }
 
+/**
+ * Build a Promise that will resolve after the specified number of milliseconds.
+ * @param ms milliseconds to wait
+ * @param value value for eventual resolution
+ */
+export function delay<T>(ms: number, value?: T): Promise<T> {
+  return new Promise((resolve) => {
+    setTimeout(() => resolve(value), ms);
+  });
+}
+
+/**
+ * Reduce an asynchronous iterable into a single value.
+ * @param iterable the async iterable to be reduced
+ * @param callbackfn a function that implements one step of the reduction
+ * @param initialValue the initial value for the accumulator
+ */
+export async function awaitAndReduce<T, U>(iterable: AsyncIterable<T>,
+                                           callbackfn: (previousValue: U, currentValue: T) => U,
+                                           initialValue: U): Promise<U> {
+  // TODO: make initialValue optional (overloads or conditional types?)
```

Comment:
🤔

Reply:
meh, i think i'll leave that comment in there.

right now we don't need a version of this function that doesn't take an `initialValue` because i'm manually passing `{} as WebAPICallResult` into it. but, since this lives in `util.ts`, it would be more reusable if it didn't need an initial value, the way that [Array.prototype.reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) also doesn't require an initial value.